src/compiler/types/stencil-types.ts

```diff
@@ -70,9 +74,9 @@ export const updateTypeIdentifierNames = (
  * @param sourceFilePath the component source file path to resolve against
  * @returns the path of the type import
  */
-const getTypeImportPath = (importResolvedFile: string | undefined, sourceFilePath: string): string => {
-  const isPathRelative = importResolvedFile && importResolvedFile.startsWith('.');
-  if (isPathRelative) {
+const getTypeImportPath = (importResolvedFile: string | undefined, sourceFilePath: string): string | undefined => {
```

Comment:
This fn could always return `undefined`, especially if `importResolvedFile` was falsy - this makes that explicit (and something we end up dealing with right after its [called here](https://github.com/ionic-team/stencil/pull/5026/files#diff-9520ec3dc9b12aac1a3735d5bc8a2808c1cf32a7c560ab81874b9e6aa10b9f37R56-R59))